src/SourceFile.js

```diff
@@ -0,0 +1,229 @@
+     */
+    getRaw() {
+        if (this.raw === undefined) {
+            this.read();
+        }
+        return this.raw;
+    }
+
+    /**
+     * The content of the file, stored as regular Javascript string
+     * encoded in UTF-8.
+     *
+     * @type {String}
+     * @protected
+     */
+    #content;
+    
+    /**
+     * Get the content of this file encoded as a regular Javascript
+     * string.
+     * @returns {String} the content of the file, encoded as a JS string
+     */
+    getContent() {
+        if (this.content === undefined) {
+            this.read();
+        }
+        return this.content;
+    }
+    
+    /**
+     * Get the content of this file, encoded as an array of lines. Each
+     * line has its trailing newline character removed.
+     * @returns {Array.<String>} the content as an array of lines
+     */
+    getLines() {
+        if (this.content === undefined) {
+            this.read();
+        }
+        return this.content.split(/\n/g);
+    }
+
+    /**
+     * Set the content of this file to the given array of lines. Each
+     * line should not have a trailing newline character.
+     * @param {Array.<String>} lines the lines to set as the content
+     */
+    setLines(lines) {
+        if (Array.isArray(lines)) {
+            this.content = lines.join('\n');
+            this.dirty = true;
+        }
+    }
+
+    /**
+     * The current length of the file, including any modifications.
+     * @returns {Number} the length in Unicode characters of this file
+     */
+    getLength() {
+        if (this.content === undefined) {
+            this.read();
+        }
+        return this.content.length;
+    }
+
+    /**
+     * The type of this file.
+     * @type {String}
+     * @protected
+     */
+    #type;
+    
+    /**
+     * Return the type of this file.
+     * @returns {String} the type of this file
+     */
+    getType() {
+        return this.type;
+    }
+
+    /**
+     * Whether or not the file has been modified.
+     * @type {Boolean}
+     * @protected
+     */
+    #dirty = false;
+
+    /**
+     * Return whether or not this instance has been modifed
+     * from the original source.
+     * 
+     * @returns {Boolean} true if the file has been modified
+     */
+    isDirty() {
+        return this.dirty;
+    }
+
+    /**
+     * Source locale for this file.
+     * @readonly
+     * @type {string}
+     * @protected
+     */
+    #sourceLocale;
+
+    /**
+     * Get the source locale for this rule.
+     *
+     * @returns {String} the source locale for this rule
+     */
+    getSourceLocale() {
+        return this.sourceLocale;
+    }
+
+    /**
+     * Write the file. If the path is given and it is different from the
+     * path of the current file, then it is written to the other path.
+     * Otherwise, this file will overwrite the source file. The path to
+     * the file must exist first.
+     *
+     * @param {String} [filePath] optional path to write the file to
```

Comment:
Why support writing to a different location here? Regardless, such write should update the path property

Reply:
I planned to have a command-line option where you can have it write the auto-fixes to a parallel file instead of modifying the original source file.